### ISSUE TITLE
The thread is blocked after libevent calls the bufferevent_write created by async.

### ISSUE THREAD

**Issue by TheDragonK** [libevent/libevent#1122]
```
void *threadFn()
{
    string retStr = "woshoudaol$_S_H";
    bufferevent *dev = AI_Server::aiClientMap["123"];
    if (NULL != dev)
    {
        bufferevent_write(dev, retStr.c_str(), strlen(retStr.c_str()));
        printf("%s", "111");
    }
}

int test::sendMsg(bufferevent *dev)
{
    std::async(std::launch::async, threadFn);
    // std::thread task(threadFn);
    // task.detach();
}

**Comment by azat** [comment]
bufferevent_write is async, it only adds data to the buffer, however it may have a lock (with BEV_OPT_THREADSAFE), which will be acquired in write code path.

From your lines of code it is hard to say, can you provide a reproducer?
Also can you attach with gdb to the program when it hang and attach stacktrace to this issue?

**Comment by azat** [comment]
Closing, due to lack of reply, feel free to reopen
